File: src/main/java/com/khg/info_master/domain/Question.java
package com.khg.info_master.domain;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "question")
public class Question {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer exam_year;
    private Integer round;
    private Integer number;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String questionText;

    private String difficulty;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tag_id", nullable = false)
    private Tag tag;

    @OneToOne(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true)
    private Answer answer;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: info-master-front/src/types/Question.ts
import type  { Answer } from "./Answer";

// 서버에서 내려오는 응답 DTO
export interface Question {
  id: number;
  exam_year: number;
  round: number;
  number: number;
  questionText: string;
  difficulty?: string;

  tagId: number;
  tagName: string;

  // 🔥 추가: 답안 (없을 수도 있음)
  answer?: Answer | null;

  createdAt?: string;
  updatedAt?: string;
}


// API 요청 전용 DTO 타입
// 질문 생성 및 수정 시에 사용
export interface QuestionCreate {
  exam_year: number;
  round: number;
  number: number;
  questionText: string;
  tagId: number;
  difficulty?: string;
}
